src/js/legend.js

```diff
@@ -273,6 +273,7 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
         that.events.onLegendCreated.fire();
     };
 
+    // TODO: refactor into a utility function, now used by sonifier
```

Comment:
I think it makes sense to create a separate ```utils.js``` file. As it will be under ```chartAuthoring``` folder, using ```utils``` should be ok. 

I'd prefer still keep d3 related operations in a separate file so that we know the dependency. If someday we decided to use another drawing library, it saves us from the cherry picking in the code.